### PR TITLE
Added single-thread Heron Instance in python

### PR DESCRIPTION
**Pull request by taishi8117**
``st_heron_instance.py`` is an implementation of Heron Instance with single-thread in Python. They server similar functionalities as ``HeronInstance``, ``Gateway``, ``Slave`` etc. in Java implementation.

[Changes](https://github.com/twitter/heron/pull/1201/files#diff-3e725585fbf57518c1e7209b1b18e401R13) to ``heron_core.bzl`` is necessary to include this heron instance to Heron's core package. Please refer to #1195 as well.

This PR depends on #1236

### CODE REVIEW

File: heron/instance/src/python/instance/BUILD [hunk 1]
```diff
@@ -0,0 +1,17 @@
+package(default_visibility = ["//visibility:public"])
+
+load("/tools/rules/pex_rules", "pex_library", "pex_binary")
+
+pex_binary(
+  # build binary for single thread python heron instance
+  name = "heron-python-instance",
+  srcs = ["st_heron_instance.py"],
```
Comment (kramasamy) [inline heron/instance/src/python/instance/BUILD]: why call st_heron_instance? simply calling heron_instance will suffice?

File: heron/instance/src/python/instance/__init__.py [hunk 2]
```diff
@@ -0,0 +1,4 @@
+'''single thread heron instance module'''
+__all__ = ['st_heron_instance.py', 'st_stmgr_client.py']
```
Comment (kramasamy) [inline heron/instance/src/python/instance/__init__.py]: same as before heron_instance.py and stmgr_client.py

File: heron/instance/src/python/instance/st_heron_instance.py [hunk 3]
```diff
@@ -0,0 +1,269 @@
...
+
+    self.in_stream = HeronCommunicator(producer_cb=None, consumer_cb=None)
+    self.out_stream = HeronCommunicator(producer_cb=None, consumer_cb=None)
+
+    self.socket_map = dict()
+    self.looper = GatewayLooper(self.socket_map)
+
+    # Initialize metrics related
+    self.out_metrics = HeronCommunicator()
+    self.out_metrics.\
```
Comment (kramasamy) [inline heron/instance/src/python/instance/st_heron_instance.py]: can you break the line in the argument as compared to in the middle of invocation?

File: heron/instance/src/python/instance/BUILD (hunk 1 above)
Comment (taishi8117) [inline heron/instance/src/python/instance/BUILD]: @kramasamy I just thought if we are going to add double-thread heron instance later, it's better to distinguish between them.

Comment (kramasamy) [inline heron/instance/src/python/instance/BUILD]: ok.

**Pull request merged**